src/test/java/io/sniffy/socket/QueryCounterSocketExpectationsTest.java

```diff
@@ -0,0 +1,66 @@
+package io.sniffy.socket;
+
+import io.sniffy.*;
+import io.sniffy.junit.QueryCounter;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import java.io.IOException;
+
+/**
+ * Created by bedrin on 01.05.2016.
+ */
+public class QueryCounterSocketExpectationsTest extends BaseSocketTest {
+
+    @Rule
+    public final ExpectedException thrown = ExpectedException.none();
+
+    @Rule
+    public final QueryCounter queryCounter = new QueryCounter();
+
+    // TODO: make autodiscoverable
+    @Before
+    public void installSocketFactory() throws IOException {
+        SnifferSocketImplFactory.install();
+    }
+
+    @After
+    public void uninstallSocketFactory() {
+        SnifferSocketImplFactory.uninstall();
+    }
+
+    @Test
+    @SocketExpectation(connections = @Count(2))
+    public void testExactConnections() {
+        performSocketOperation();
+        performSocketOperation();
+    }
+
+    @Test
+    @SocketExpectation(connections = @Count(3))
+    public void testExactConnections_Failed() {
+        performSocketOperation();
+        performSocketOperation();
+        thrown.expect(SniffyAssertionError.class);
+    }
+
+    @Test
+    @SocketExpectation(connections = @Count(2))
+    public void testMinConnections() {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [JUnit tests should include assert() or fail()](https://www.codacy.com/app/sniffy/sniffy/file/2496541651/issues/source?bid=3005960&fileBranchId=3034483#l52)